frontend/src/HomePage/FolderFilter/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { isEmpty } from 'lodash';
+import { useNavigate } from 'react-router-dom';
+import Select from 'react-select';
+
+export default function FolderFilter({ disabled, options, onChange, value }) {
+  const navigate = useNavigate();
+  const [currentFolder, setCurrentFolder] = useState({});
+  let selectRef = useRef();
+
+  const handleFolderChange = (folder) => {
+    onChange(folder);
+    updateFolderQuery(folder?.name, folder?.id);
+  };
+
+  const updateFolderQuery = (name, id) => {
+    const path = `${id ? `?folder=${name}` : ''}`;
+    navigate({ pathname: location.pathname, search: path }, { replace: true });
+  };
+
+  useEffect(() => {
+    if (isEmpty(value)) {
+      setCurrentFolder({ name: 'All apps', label: 'All apps', value: '', id: '' });
```

Comment:
Saw this obj is using in 2 places. shall we convert it to a variable and reuse it?
```suggestion
     const defaultFolder = { name: 'All apps', label: 'All apps', value: '', id: '' }
      setCurrentFolder(defaultFolder);
```